presto-orc/src/main/java/com/facebook/presto/orc/reader/ByteSelectiveStreamReader.java

```diff
@@ -258,18 +262,37 @@ private int readNoFilter(int[] positions, int positionCount)
             throws IOException
     {
         // filter == null implies outputRequired == true
-        if (positions[positionCount - 1] == positionCount - 1) {
+        int maxPositionCount = positions[positionCount - 1] + 1;
+
+        if (maxPositionCount * (1 - BATCH_EXECUTION_FILTERRATE_THRESHOLD) <= positionCount) {
```

Comment:
consider moving this into a helper method in ReaderUtils

Reply:
> consider moving this into a helper method in ReaderUtils

I added a variable `useBatchRead` in the next PR and set it in read() before ensureCapacity. But then I realized the condition to use batch read may not be the same for different cases. For example, for the with filter case, no null is always better than with null when using batch mode. And for different types, this threshold may be different. Let me do a few more types and then decide how to do this.